Cape Town | May-2025 | Tinotenda Mnyamazi | Wireframe

Tinotenda55 opened the pull request:
<!--

You must title your PR like this:

REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME

For example,

London | May-2025 | Carol Owen | Wireframe

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist

Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

Tinotenda55 commented:
i believe that i did everything that was task for the project

cjyuan commented on Wireframe/index.html:
```diff
@@ -6,28 +6,52 @@
+
+      <div class="article3">
+        <img src="./assets/git-branch.jpg" id="3" />
+        <h2>What is a branch in Git</h2>
```
I think this heading can also end with a question mark (to be consistent with the other two headings).

cjyuan reviewed (commented):
Can you address these issues (in addition to the suggestions made in the code)?

1. There are errors in your HTML code.

2. Lighthouse Accessibility score is not yet 100.

3. When a wireframe is provided, our implementation should closely reflect its appearance and layout to ensure consistency with design expectations. You're off to a solid start! To better align with the wireframe, here are a few areas where the layout and appearance could be refined:
    - The right edge of the 1st article does not quite align with the right edge of the 3rd article.
    - The 2nd and 3rd articles appear to have different widths.

4. One of the acceptance criteria, "The page footer is fixed to the bottom of the viewport", has not yet been satisfied. Can you make the necessary change? (Suggestion: Ask ChatGPT what that requirement means).


On a separate note, it is best practice to include a brief description of the changes you made in the Changelist section of every PR.

Tinotenda55 commented:
i have made a number of changes and i commit with a message mistakenly , and I run the lighthouse  Accessibility score is now 100

cjyuan commented on Wireframe/index.html:
```diff
@@ -6,28 +6,52 @@
-    </footer>
+        <button href="https://bradfrost.com/blog/post/html-wireframes/">
+          READ MORE
+        </button>
```
It's a common practice to respond to individual comments. For example, you can say, "I have changed the button to an `<a>` element".

cjyuan commented:
Your implementation is getting better and the page looks more like the wireframe. Good job.

1. To even better align with the wireframe, here are a few areas where the layout and appearance could be refined:
    - In the wireframe, the width of the images in the 2nd and 3rd articles spans the whole article.
    - In the wireframe, each "Read me" link has a border. Can you introduce border to the links in your implementation?

2. One of the acceptance criteria, "The page footer is fixed to the bottom of the **viewport**", has not yet been satisfied. Can you make the necessary change? (Suggestion: Ask ChatGPT what that requirement means).

3. ChatGPT showed me some useful suggestions and pointed out some typos and grammar issues on your page when I asked it this:
```
review HTML code:
...        <--- paste your HTML code here
```

To get better at using AI tools, can you try running your code by ChatGPT? Then take a look at the suggestions it gives you, and pick out the ones that actually make sense for your project. It's a great way to learn and improve your work.

---
On a separate note, in the CYF course, the recommended way to let a reviewer know you've made the requested changes is to:
  - Reply to their comments (which you’ve already done — great!)
  - Replace the "Reviewed" label by a "Needs review" label.

Tinotenda55 commented:
> Please don't request for reviewer.
> 
> > In the CYF course, the recommended way to let a reviewer know you've made the requested changes is to:
> > 
> > * Reply to their comments
> > * Replace the "Reviewed" label by a "Needs review" label.
> 
> Can you describe what you have changed since the last review? Can you update the label to indicate that your changes are ready to be reviewed?

noted 
. I have change the id name of article 1 ,2 and 3 images 
. Added a <main> on my html 
. Added width of 100% on .article2 image and .article3 image 
. Added auto height on article2 and 3 images

cjyuan reviewed (commented):
1. There are still HTML errors in your code.

2. The main purpose of implementing a wireframe is to translate structure and functionality, not final visual design. Visual design is usually handled by UI/graphic designers after the wireframe phase. To further better align with the wireframe, here are a few areas where the layout and appearance could be refined:
    - In the wireframe, the images in the 2nd and 3rd articles have the same height.
    - The "Read me" links have a visible border.
    - Choice of image: the image in the 3rd articles comes with a border, making it difficult to tell if the border was added using CSS or not. (For wireframe implementation, you can actually use `placeholder.svg` for all three articles.)
 ![image](https://github.com/user-attachments/assets/0640d6ca-de96-4529-84d8-bfee75441af5)
 
3. One of the acceptance criteria, "The page footer is fixed to the bottom of the viewport", has not yet been satisfied. Can you make the necessary change? (Suggestion: Ask ChatGPT what that requirement means).

4. One of the learning objectives mentioned in the `README.md` is "Use semantic HTML tags to structure the webpage". Can you find out from ChatGPT how you can modify your HTML code to enhance its semantic structure? In particular, find out you can replace some of your `<div>` elements by semantic elements such as `<main>`, `<header>`, `<article>`, `<footer>`.

5. There are spelling mistakes in the text content. 

6. A clean branch is a branch that contains only relevant changes for a specific task or feature (e.g., this Wireframe exercise), with no unrelated edits — making it easy to review, test, and merge.
    - Is `Wireframe/DET Guide 2024-10-28 EN.pdf` part of your Wireframe implementation? (If not, please delete this file)
    - Is `Form-Controls/index.html` part of your Wireframe implementation? (If not, please replace this file by the same file in the `main` branch)

cjyuan commented:
I don't see any new change since my previous review.
If you have made any change since my last review, have you pushed the changes to Github?

Tinotenda55 commented:
I have done correcting and upgrading all that have been marked down on the last comment i have receive , i would like to know is they is still more .

cjyuan commented:
"Form-Controls/index.html" in this branch is different from "Form-Controls/index.html" in the `main` branch.

By "original form", I mean the the state of the file in the `main` branch.

cjyuan commented:
Your recent commit did not change anything.

To restore the `Form-Controls/index.html` in this branch, replace this file (or replace its content) in this branch by the same file in your `main` branch: https://github.com/Tinotenda55/Module-Onboarding/blob/main/Form-Controls/index.html

Tinotenda55 commented:
i have got to understand what you have been say i about the restoration thanks @ @cjyuan

cjyuan reviewed (commented):
You have made a significant improvement to this PR. Great job!
I will mark go ahead and mark this PR as "Complete".

Please note that for best practice, we should give a brief description (under the "Changelist" section) summarizing the purpose of the PR and the changes we’ve made.
![image](https://github.com/user-attachments/assets/38fa0c46-98fc-44db-8c63-126374208fa4)
